lib/honeybadger/backend/base.rb

```diff
@@ -109,6 +109,17 @@ def track_deployment(payload)
         notify(:deploys, payload)
       end
 
+      # Send event
+      # @example
+      #   backend.event("email_received", "2023-03-04T12:12:00+1:00", { subject: 'Re: Aquisition' })
+      #
+      # @param [String] event_type Type of event to send
+      # @param [String] timestamp RFC3339 conforming formatted timestamp
+      # @param [Hash]   payload Generic payload
+      def event(event_type, timestamp, payload = {})
```

Comment:
tbh I think it would better to accept a fully constructed payload as the argument, like the other methods do.

Reply:
Something like this:

```ruby
# base
def event(payload)
  raise NotImplementedError, "must define #event on subclass"
end

# debug
def event(payload)
  logger.unknown("sending event to debug backend with event=#{payload.to_json}")
  return Response.new(ENV['DEBUG_BACKEND_STATUS'].to_i, nil) if ENV['DEBUG_BACKEND_STATUS']
  super
end

# null
def event(payload)
  StubbedResponse.new
end

# server
def event(payload)
  delimited_json = Array(payload).map(&:to_json).join("\n")
  # http.post(delimited_json)
end
```

Reply:
Yeah, on second thought this makes more sense. 

I'll rejigger this today while we're discussing the backend/worker stuff.

Reply:
Looks good!